lib/msal-node/src/cache/distributed/DistributedCachePlugin.ts

```diff
@@ -32,25 +32,34 @@ export class DistributedCachePlugin implements ICachePlugin {
     public async afterCacheAccess(
         cacheContext: TokenCacheContext
     ): Promise<void> {
-        if (cacheContext.cacheHasChanged) {
-            const kvStore = (
-                cacheContext.tokenCache as TokenCache
-            ).getKVStore();
-            const accountEntities = Object.values(kvStore).filter((value) =>
-                AccountEntity.isAccountEntity(value as object)
-            );
-
-            if (accountEntities.length > 0) {
-                const accountEntity = accountEntities[0] as AccountEntity;
-                const partitionKey = await this.partitionManager.extractKey(
-                    accountEntity
-                );
-
-                await this.client.set(
-                    partitionKey,
-                    cacheContext.tokenCache.serialize()
-                );
```

Comment:
Instead of refactoring everything, could you keep this as simple as possible? Lets move this out of the if statement (after it), and add an else for ```partitionKey = this.partitionManager.getKey()```

Reply:
Do you mean something like that?

```sh
let partitionKey;

if (accountEntities.length > 0) {
  const accountEntity = accountEntities[0] as AccountEntity;
  
  partitionKey = await this.partitionManager.extractKey(
    accountEntity
  );
} else {
  partitionKey = this.partitionManager.getKey()
}

await this.client.set(
  partitionKey,
  cacheContext.tokenCache.serialize()
);
```